src/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable_factory.js

```diff
@@ -1,101 +1,59 @@
 import $ from 'jquery';
 import { getVisualizeLoader } from 'ui/visualize/loader';
-
-import { UtilsBrushEventProvider as utilsBrushEventProvider } from 'ui/utils/brush_event';
-import { FilterBarClickHandlerProvider as filterBarClickHandlerProvider } from 'ui/filter_bar/filter_bar_click_handler';
 import { EmbeddableFactory, Embeddable } from 'ui/embeddable';
-import { PersistedState } from 'ui/persisted_state';
+import { VisualizeEmbeddable } from './visualize_embeddable';
 
 import labDisabledTemplate from './visualize_lab_disabled.html';
 
-import chrome from 'ui/chrome';
-
 export class VisualizeEmbeddableFactory extends EmbeddableFactory {
-  constructor(savedVisualizations, timefilter, Promise, Private, config) {
+  constructor(savedVisualizations, Promise, config) {
     super();
     this._config = config;
     this.savedVisualizations = savedVisualizations;
     this.name = 'visualization';
     this.Promise = Promise;
-    this.brushEvent = utilsBrushEventProvider(timefilter);
-    this.filterBarClickHandler = filterBarClickHandlerProvider(Private);
   }
 
   getEditPath(panelId) {
     return this.savedVisualizations.urlFor(panelId);
   }
 
-  render(domNode, panel, container) {
-    const editUrl = this.getEditPath(panel.id);
-
-    const waitFor = [ getVisualizeLoader(), this.savedVisualizations.get(panel.id) ];
+  /**
+   *
+   * @param {Object} panelMetadata. Currently just passing in panelState but it's more than we need, so we should
+   * decouple this to only include data given to us from the embeddable when it's added to the dashboard. Generally
+   * will be just the object id, but could be anything depending on the plugin.
+   * @param {function} onEmbeddableStateChanged
+   * @return {Promise.<{ metadata, onContainerStateChanged, render, destroy }>}
+   */
+  create(panelMetadata, onEmbeddableStateChanged) {
+    const visId = panelMetadata.id;
+    const editUrl = this.getEditPath(visId);
+
+    const waitFor = [ getVisualizeLoader(), this.savedVisualizations.get(visId) ];
     return this.Promise.all(waitFor)
```

Comment:
Great point. We should be using standard promises, though historically it's been angular promises everywhere.  As of  https://github.com/elastic/kibana/issues/13855 we want to use native.  Nathan actually has a WIP PR  that removes these, but I can take care of it here too (since this is all refactored anyway).